js/projects/projectsView.js

```diff
@@ -1,76 +1,80 @@
-  });
-};
+  projectsView.handleProjectsFilter = function() {
+    $('#type-filter').on('change', function(){
+      var $projects = $('article');
+      var $value = $(this).val();
+      if ($value === 'show-all') {
+        $projects.show();
+      } else if ($value) {
+        $projects.hide();
+        $projects.each(function(){
+          if ($(this).data('type') === $value) {
+            $(this).show();
+          }
+        });
+      } else {
+        $projects.show();
+      }
+      $('#type-filter').val('');
+    });
+  };
 
-projectsView.handleMainNav = function() {
+  projectsView.handleMainNav = function() {
 
-  $('#pageHeader nav').on('click', 'li', function(){
-    var $value = $(this).attr('data-content');
-    if (!($(this).find('a').text() === 'Github')) {
-      $('.tab-content').hide();
-      $('.tab-content').each(function(){
-        if($(this).attr('id') === $value){
-          $(this).show();
-        }
-      });
-    }
-  });
+    $('#pageHeader nav').on('click', 'li', function(){
+      var $value = $(this).attr('data-content');
+      if (!($(this).find('a').text() === 'Github')) {
+        $('.tab-content').hide();
+        $('.tab-content').each(function(){
+          if($(this).attr('id') === $value){
+            $(this).show();
+          }
+        });
+      }
+    });
 
-};
+  };
 
-projectsView.handleTruncatedDescription = function() {
-  var minimizedElements = $('.project-description p');
-  var minimizeCharacterCount = 100;
+  projectsView.handleTruncatedDescription = function() {
+    var minimizedElements = $('.project-description p');
+    var minimizeCharacterCount = 100;
 
-  minimizedElements.each(function(){
-    var t = $(this).text();
-    if(t.length < minimizeCharacterCount ) return;
+    minimizedElements.each(function(){
+      var t = $(this).text();
+      if(t.length < minimizeCharacterCount ) return;
 
-    $(this).html(
-            t.slice(0,minimizeCharacterCount )+'<span>... </span><a href="#" class="more">More</a>'+
-            '<span class="less">'+ t.slice(minimizeCharacterCount ,t.length)+' <a href="#" class="less">Less</a></span>'
-        );
-  });
+      $(this).html(
+              t.slice(0,minimizeCharacterCount )+'<span>... </span><a href="#" class="more">More</a>'+
+              '<span class="less">'+ t.slice(minimizeCharacterCount ,t.length)+' <a href="#" class="less">Less</a></span>'
+          );
+    });
 
-  $('a.more', minimizedElements).click(function(event){
-    event.preventDefault();
-    $(this).hide().prev().hide();
-    $(this).next().show();
-  });
+    $('a.more', minimizedElements).click(function(event){
+      event.preventDefault();
+      $(this).hide().prev().hide();
+      $(this).next().show();
+    });
 
-  $('a.less', minimizedElements).click(function(event){
-    event.preventDefault();
-    $(this).parent().hide().prev().show().prev().show();
-  });
-};
+    $('a.less', minimizedElements).click(function(event){
```

Comment:
Update click event to the .on event handler. Also see if there is any way to simplify dom traversal methods.